File: Apps/BankCore/backend/src/main/java/org/d3javu/backend/model/business/invoice/Invoice.java
package org.d3javu.backend.model.business.invoice;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.d3javu.backend.model.business.paymentaccount.PaymentAccount;

import java.util.UUID;

@Entity
@Table(name = "invoice")
@Getter
@Setter
@NoArgsConstructor
public class Invoice {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "amount")
    private Double amount;

    @ManyToOne
    @JoinColumn(name = "payment_account_id")
    private PaymentAccount providerPaymentAccount;

}

File: Apps/BankCore/backend/src/main/scala/domain/business/Invoice.scala
package org.d3javu
package domain.business

import io.estatico.newtype.macros.newtype
import org.d3javu.domain.business.Invoice._

import java.util.UUID

case class Invoice(
                  id: Id,
                  amount: Amount,
                  providerPaymentAccount: PaymentAccount.Id
                  )

object Invoice {

  @newtype
  final case class Id(asUUID: UUID)

  @newtype
  final case class Amount(asDouble: Double)

}
